### PR TITLE
scanmanga novel addition

### LINKED ISSUE #2557: [Site Request] Scan Manga novels
**Name of the website**
Scan Manga

**Website urls (examples below)**
- Site: https://www.scan-manga.com/?home
- Manga List: 
- Manga example: I Alone Level-Up (Novel)
- Chapter online viewer example: https://www.scan-manga.com/lecture-en-ligne/I-Alone-Level-Up-Novel-SL-Novel-1-Chapitre-1-FR_93096.html

**Languages**
French

**Additional details**
This site is already have a connector but it don't work on LN (Light Novel)

### PR DESCRIPTION
**Pull request by Robonau**
resolves #2557
not much of this site is actually functional anymore, many in the manga list don't work at all and some work on HN when they don't on the site.

### CODE REVIEW

File: src/web/mjs/connectors/ScanManga.mjs [hunk 1]
```diff
@@ -72,6 +72,32 @@ export default class ScanManga extends Connector {
...
+        return data.length > 0 ? this._getPagesNovel(request) : this._getPagesManga(chapter);
+    }
+
+    async _getPagesNovel(request) {
+        let script = `
+            new Promise(resolve => {
+                document.body.style.width = '56em';
+                let novel = document.querySelector('article.aLN');
+                novel.style.padding = '1.5em';
+                let script = document.createElement('script');
```
Comment (27136773) [inline src/web/mjs/connectors/ScanManga.mjs:89]: May add a `script.onerror = error => reject(error);` to ensure promise is rejected immediately when loading script failed

File: src/web/mjs/connectors/ScanManga.mjs [hunk 2]
```diff
@@ -72,6 +72,32 @@ export default class ScanManga extends Connector {
...
+                script.onload = async function() {
+                    let canvas = await html2canvas(novel);
```
Comment (27136773) [inline src/web/mjs/connectors/ScanManga.mjs]: May wrap method body in `try ... catch ...` to process any unexpected error and reject the promise immediately

**Review by 27136773 — APPROVED** [review]
Looks good

File: src/web/mjs/connectors/ScanManga.mjs [hunk 3]
```diff
@@ -72,6 +72,37 @@ export default class ScanManga extends Connector {
...
+        let uri = new URL(chapter.id, this.url);
+        uri.searchParams.set('style', 'list');
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, '.aLN');
+        return data.length > 0 ? this._getPagesNovel(request) : this._getPagesManga(chapter);
+    }
+
+    async _getPagesNovel(request) {
+        let script = `
+            new Promise(resolve => {
```
Comment (27136773) [inline src/web/mjs/connectors/ScanManga.mjs]: When using reject, it must be defined in the callback of the promise as parameter
`new Promise((resolve, reject) => { ... });`

**Comment by Robonau** [comment]
should madara novel and the recently changed kol novel be changed too?
edit: did it anyways

**Review by 27136773 — APPROVED** [review]
Looks good

**Comment by 27136773** [comment]
Resolve conflict from merge of previous PR

**Comment by Robonau** [comment]
welp the conflict is fixed

**Pull request merged**